frontend/packages/kubevirt-plugin/src/selectors/vm/selectors.ts

```diff
@@ -53,51 +54,13 @@ export const getVolumes = (vm: VMKind, defaultValue: V1Volume[] = []): V1Volume[
 export const getDataVolumeTemplates = (vm: VMKind, defaultValue = []) =>
   _.get(vm, 'spec.dataVolumeTemplates') == null ? defaultValue : vm.spec.dataVolumeTemplates;
 
-export const getConfigMapVolumes = (vm: VMKind, defaultValue: V1Volume[] = []): V1Volume[] =>
-  getVolumes(vm, defaultValue).filter((vol) => Object.keys(vol).includes('configMap'));
-
-export const getSecretVolumes = (vm: VMKind, defaultValue: V1Volume[] = []): V1Volume[] =>
-  getVolumes(vm, defaultValue).filter((vol) => Object.keys(vol).includes('secret'));
-
-export const getServiceAccountVolumes = (vm: VMKind, defaultValue: V1Volume[] = []): V1Volume[] =>
-  getVolumes(vm, defaultValue).filter((vol) => Object.keys(vol).includes('serviceAccount'));
-
-export const getEnvDiskVolumes = (vm: VMKind, defaultValue: V1Volume[] = []): V1Volume[] => [
-  ...getConfigMapVolumes(vm, defaultValue),
-  ...getSecretVolumes(vm, defaultValue),
-  ...getServiceAccountVolumes(vm, defaultValue),
-];
-
-export const getConfigMapDisks = (vm: VMKind, defaultValue: V1Disk[] = []): V1Disk[] =>
-  getDisks(vm, defaultValue).filter(
-    (disk) => !!getConfigMapVolumes(vm).find((vol) => vol.name === disk.name),
-  );
-
-export const getSecretDisks = (vm: VMKind, defaultValue: V1Disk[] = []): V1Disk[] =>
-  getDisks(vm, defaultValue).filter(
-    (disk) => !!getSecretVolumes(vm).find((vol) => vol.name === disk.name),
-  );
-
-export const getServiceAccountDisks = (vm: VMKind, defaultValue: V1Disk[] = []): V1Disk[] =>
-  getDisks(vm, defaultValue).filter(
-    (disk) => !!getServiceAccountVolumes(vm).find((vol) => vol.name === disk.name),
-  );
-
-export const getEnvDisks = (vm: VMKind, defaultValue: V1Disk[] = []): V1Disk[] => [
-  ...getConfigMapDisks(vm, defaultValue),
-  ...getSecretDisks(vm, defaultValue),
-  ...getServiceAccountDisks(vm, defaultValue),
-];
-
-export const getBootableDisks = (vm: VMKind, defaultValue: V1Disk[] = []): V1Disk[] =>
-  getDisks(vm, defaultValue).filter(
-    (disk) => !getEnvDisks(vm).find((envDisk) => envDisk.name === disk.name),
-  );
-
-export const getNonBootableDisks = (vm: VMKind, defaultValue: V1Disk[] = []): V1Disk[] =>
-  getDisks(vm, defaultValue).filter((disk) =>
-    getEnvDisks(vm).find((envDisk) => envDisk.name === disk.name),
-  );
+export const getBootableDisks = (vm: VMKind, defaultValue: V1Disk[] = []): V1Disk[] => {
+  const volumeLookup = createBasicLookup(getVolumes(vm), getSimpleName);
+  return getDisks(vm, defaultValue).filter((disk) => {
+    const volWrapper = new VolumeWrapper(volumeLookup[disk.name]);
+    return !volWrapper.isEmpty() && volWrapper.getType() && !volWrapper.getType().isEnvironmentType;
```

Comment:
it is better to have a method instead of just isEnvironmentType field. 

The field is sort of an implementation detail and a method can be extended more easily in the future.

Reply:
+1

Reply:
you can use it inside the class and expose just the method

Reply:
it is not a big issue here; just a nitpick